src/server/scripts/EasternKingdoms/ZulAman/boss_akilzon.cpp

```diff
@@ -97,32 +75,103 @@ class boss_akilzon : public CreatureScript
-            events.ScheduleEvent(EVENT_CALL_LIGHTNING, urand(10000, 20000));    // totaly random timer. can't find any info on this
-            events.ScheduleEvent(EVENT_ELECTRICAL_STORM, 60000);                // 60 seconds(bosskillers)
-            events.ScheduleEvent(EVENT_RAIN, urand(47000, 52000));
-            events.ScheduleEvent(EVENT_ENRAGE, 10 * MINUTE * IN_MILLISECONDS);  // 10 minutes till enrage(bosskillers)
+            ScheduleTimedEvent(10s, 20s, [&]{
+                Unit* target = SelectTarget(SelectTargetMethod::Random, 1);
+                if (!target)
+                    target = me->GetVictim();
+                if (target)
+                {
+                    TargetGUID = target->GetGUID();
+                    DoCast(target, SPELL_STATIC_DISRUPTION, false);
+                    me->SetInFront(me->GetVictim());
+                }
+            }, 10s, 18s);
+
+            ScheduleTimedEvent(20s, 30s, [&] {
+                Unit* target = SelectTarget(SelectTargetMethod::Random, 1);
+                if (!target)
+                    target = me->GetVictim();
+                if (target)
+                    DoCast(target, SPELL_GUST_OF_WIND);
+            }, 20s, 30s);
+
+            ScheduleTimedEvent(10s, 20s, [&] {
+                DoCastVictim(SPELL_CALL_LIGHTNING);
+            }, 12s, 17s);
+
+            ScheduleTimedEvent(1min, [&] {
+                Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 50, true);
+                if (!target)
+                {
+                    EnterEvadeMode();
+                    return;
+                }
+                target->CastSpell(target, 44007, true); // cloud visual
+                DoCast(target, SPELL_ELECTRICAL_STORM, false); // storm cyclon + visual
+                float x, y, z;
+                target->GetPosition(x, y, z);
+
+                Unit* Cloud = me->SummonTrigger(x, y, me->GetPositionZ() + 16, 0, 15000);
+                if (Cloud)
+                {
+                    target->GetMotionMaster()->MoveJump(Cloud->GetPosition(), 1.0f, 1.0f);
+
+                    CloudGUID = Cloud->GetGUID();
+                    Cloud->SetDisableGravity(true);
+                    Cloud->StopMoving();
+                    Cloud->SetObjectScale(1.0f);
+                    Cloud->SetFaction(FACTION_FRIENDLY);
+                    Cloud->SetMaxHealth(9999999);
+                    Cloud->SetHealth(9999999);
+                    Cloud->RemoveUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+
+                    me->m_Events.AddEventAtOffset([&, Cloud] {
+                        HandleStormSequence(Cloud);
+                    }, 3s);
+                }
+
+                StormCount = 1;
+
+                me->m_Events.AddEventAtOffset([&] {
+                    if (!isRaining)
+                    {
+                        SetWeather(WEATHER_STATE_HEAVY_RAIN, 0.9999f);
```

Comment:
This and the other instance might need adjustment as per https://github.com/chromiecraft/chromiecraft/issues/7460#issuecomment-2444168335, likely out of scope for this PR though.

Reply:
yep, we will still return to this